pkg/daemon/writer.go

```diff
@@ -29,14 +29,16 @@ type NodeStateStatusWriter struct {
 	OnHeartbeatFailure func()
 	metaData           *utils.OSPMetaData
 	networkData        *utils.OSPNetworkData
+	devMode            bool
 }
 
 // NewNodeStateStatusWriter Create a new NodeStateStatusWriter
-func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func()) *NodeStateStatusWriter {
+func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func(), mode bool) *NodeStateStatusWriter {
```

Comment:
nit : mode -> devMode

when reading this, its hard to understand from the function signature what is a "true" or "false" mode mean.
other arguments here also suffer from lack of clarity IMO (e.g `f`)

Reply:
Done